### PR TITLE
27059 refactor dashboard cms

### PR DESCRIPTION
**Pull request by nicolaisirghi**

### CODE REVIEW

**Review by GunkaArtur — APPROVED** [review]

File: packages/demo-nextjs/src/app/api/items/latest/route.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { getItems } from "@/lib/db/item/getItems";
+import DBConnect from "@/lib/db/mongoose/connect";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function GET(req: NextRequest) {
+  try {
+    const limit = req.nextUrl.searchParams.get("limit");
+
+    await DBConnect();
```
Comment (maxval1) [inline packages/demo-nextjs/src/app/api/items/latest/route.ts:9]: remove `await DBConnect`

File: packages/demo-nextjs/src/app/api/items/latest/route.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    const { items } = await getItems({}, { sort: { createdAt: -1 }, limit: parseInt(limit ?? "5"), skip: 0 });
```
Comment (maxval1) [inline packages/demo-nextjs/src/app/api/items/latest/route.ts:11]: need to chage the interface of getItems, need to ad `sortBy: "date" | "name"` and `sort: "desc" | "asc"`

**Review by maxval1 — CHANGES_REQUESTED** [review]
conflicts

File: packages/demo-nextjs/src/lib/db/item/getItems.ts [hunk 3]
```diff
@@ -6,33 +6,47 @@ import { Item } from "../types";
 import { toItemConvertor } from "./utils";
 
 type Query = {
-  type: string;
+  type?: string;
```
Comment (maxval1) [inline packages/demo-nextjs/src/lib/db/item/getItems.ts:9]: Need to create 2 types something like this
image

**Review by maxval1 — CHANGES_REQUESTED** [review]

**Pull request merged**